src/store/channels/saga.ts

```diff
@@ -136,7 +145,63 @@ export function* roomUnfavorited(action) {
   }
 }
 
+export function* publishUserTypingEvent(roomId) {
+  try {
+    const channel = yield select(rawChannelSelector(roomId));
+    if (!channel) {
+      return;
+    }
+
+    const currentUserLastTypingEventPublishedAt = channel.currentUserLastTypingEventPublishedAt;
+    const now = Date.now();
+    if (currentUserLastTypingEventPublishedAt && now - currentUserLastTypingEventPublishedAt <= USER_TYPING_TIMEOUT) {
+      // not enough time has passed since last typing event
+      return;
+    }
+
+    yield call(matrixSendUserTypingEvent, roomId, true);
+    yield call(receiveChannel, { id: roomId, currentUserLastTypingEventPublishedAt: now });
+  } catch (error) {
+    console.error(`Failed to publish user is typing event in room ${roomId}:`, error);
+  }
+}
+
+export function* publishUserStoppedTypingEvent(roomId) {
```

Comment:
Do we actually have to do this? Does the timeout that we sent to the matrix server not just clear it after 5 seconds anyway?

Reply:
Update to only call this after message is sent